app/controllers/collections_controller.rb

```diff
@@ -189,4 +233,218 @@ def respond_with_error(message, status_code)
+        request_params[:items] = JSON.parse(request_params[:items])
+      end
+      request_params[:file] = [request_params[:file]] unless request_params[:file].nil? or request_params[:file].is_a? Array
+    end
+    request_params
+  end
+
+  # Processes files uploaded as part of a multipart request
+  def process_uploaded_files(corpus_dir, collection_name, files)
+    uploaded_files = []
+    unless files.nil?
+      files.each do |uploaded_file|
+        uploaded_files.push(upload_document_using_multipart(corpus_dir, uploaded_file.original_filename, uploaded_file, collection_name))
+      end
+    end
+    uploaded_files
+  end
+
+  # Ingests a list of items
+  def ingest_items(corpus_dir, items)
+    items_ingested = []
+    items.each do |item|
+      ingest_one(corpus_dir, item[:rdf_file])
+      items_ingested.push(item[:identifier])
+    end
+    items_ingested
+  end
+
+  # Write item JSON metadata to RDF file
+  def write_item_metadata(corpus_dir, item_json)
+    rdf_metadata = convert_json_metadata_to_rdf(item_json["metadata"])
+    rdf_file = create_item_rdf(corpus_dir, item_json["identifier"], rdf_metadata)
+    {:identifier => item_json["identifier"], :rdf_file => rdf_file}
+  end
+
+  # Validates the request on the add items api call
+  def validate_add_items_request(collection, corpus_dir, request_params)
+    validate_items(request_params[:items], collection, corpus_dir)
+    validate_files(request_params[:file], collection, corpus_dir)
+    validate_document_identifiers(get_document_identifiers(request_params[:file], request_params[:items]))
+  end
+
+  # Validates the collection exists and the user is authorised to modify it
+  def validate_collection(request_params)
+    collection = Collection.find_by_name(request_params[:id])
+    if collection.nil?
+      raise ResponseError.new(404), "Requested collection not found"
+    elsif request_params[:api_key] != User.find(collection.owner_id).authentication_token
+      raise ResponseError.new(403), "User is unauthorised" # Authorise by comparing api key sent with collection owner's api key
+    end
+    collection
+  end
+
+  # Validates the metadata for each of the items
+  def validate_items(items_metadata, collection, corpus_dir)
+    raise ResponseError.new(400), "JSON-LD formatted item metadata must be sent with the api request" if items_metadata.blank?
+    items_metadata.each do |item|
+      validate_item(item, collection, corpus_dir)
+    end
+  end
+
+  # Iterates over the uploaded files and JSON document content and returns a list of document identifiers/filenames
+  def get_document_identifiers(uploaded_files, items_metadata)
+    document_identifiers = []
+    items_metadata.each do |item|
+      unless item["documents"].nil?
+        item["documents"].each do |document|
+          document_identifiers.push(document["identifier"])
+        end
+      end
+    end
+    unless uploaded_files.nil?
```

Comment:
I don't think you need this check because `process_uploaded_files` will always return an array (possibly empty)